ci/tasks/endtoend/zap/docker-compose.yml

```diff
@@ -0,0 +1,321 @@
+    env_file:
+      - ../docker-config/java_app.env
+      - ../docker-config/ledger.env
+    environment:
+      - RUN_MIGRATION=true
+      - RUN_APP=true
+    networks:
+      pymnt_network:
+        aliases:
+          - ledger.pymnt.localdomain
+          - ledger.internal.pymnt.localdomain
+    mem_limit: 2G
+    depends_on:
+      - postgres
+      - sqs
+    healthcheck:
+      test: ["CMD", "wget", "-O", "/dev/null", "http://localhost:10700/healthcheck"]
+      interval: 1m30s
+      timeout: 10s
+    logging:
+      driver: "json-file"
+  publicauth:
+    image: ${DOCKER_REGISTRY_URI:-}${repo_publicauth:-govukpay/publicauth}:${tag_publicauth:-latest-master}
+    env_file:
+      - ../docker-config/java_app.env
+      - ../docker-config/publicauth.env
+    environment:
+      - RUN_MIGRATION=true
+      - RUN_APP=true
+    networks:
+      pymnt_network:
+        aliases:
+          - publicauth.pymnt.localdomain
+          - publicauth.internal.pymnt.localdomain
+    mem_limit: 1G
+    healthcheck:
+      test: ["CMD", "wget", "-O", "/dev/null", "http://localhost:9600/healthcheck"]
+      interval: 1m30s
+      timeout: 10s
+    depends_on:
+      - postgres
+    logging:
+      driver: "json-file"
+
+  stubs:
+    image: ${DOCKER_REGISTRY_URI:-}${repo_stubs:-govukpay/stubs}:${tag_stubs:-latest-master}
+    env_file: ../docker-config/stubs.env
+    networks:
+      pymnt_network:
+        aliases:
+          - stubs.internal.pymnt.localdomain
+    mem_limit: 1G
+    logging:
+      driver: "json-file"
+
+  stubs_proxy:
+    image: ${DOCKER_REGISTRY_URI:-}${repo_reverse_proxy:-govukpay/reverse-proxy}:latest-master
+    volumes:
+      - "./../docker-config/ssl:/ssl"
+    environment:
+      - CERT_FILE=/ssl/certs/stubs.pymnt.localdomain.crt
+      - KEY_FILE=/ssl/keys/stubs.pymnt.localdomain.key
+      - HOST=stubs.internal.pymnt.localdomain
+      - PORT=9900
+      - START_IN_DEV_MODE=false
+      - X_FRAME_OPTIONS_HEADER=ALLOW
+    networks:
+      pymnt_network:
+        aliases:
+          - stubs.pymnt.localdomain
+    mem_limit: 50M
+    logging:
+      driver: "json-file"
+
+  cardid:
+    image: ${DOCKER_REGISTRY_URI:-}${repo_cardid:-govukpay/cardid}:${tag_cardid:-latest-master}
+    env_file:
+      - ../docker-config/java_app.env
+      - ../docker-config/cardid.env
+    networks:
+      pymnt_network:
+        aliases:
+          - cardid.pymnt.localdomain
+          - cardid.internal.pymnt.localdomain
+    mem_limit: 2G
+    healthcheck:
+      test: ["CMD", "wget", "-O", "/dev/null", "http://localhost:9800/healthcheck"]
+      interval: 30s
+      timeout: 10s
+    logging:
+      driver: "json-file"
+  
+  endtoend:
+    image: ${DOCKER_REGISTRY_URI:-}${repo_endtoend:-govukpay/endtoend}:${tag_endtoend:-latest-master}
+    env_file: ../docker-config/endtoend.env
+    environment:
+      - MAVEN_OPTS=${END_TO_END_JAVA_OPTS}
+    networks:
+      - pymnt_network
+    mem_limit: ${END_TO_END_MEM_LIMIT}
+    depends_on:
```

Comment:
Probably want to add a depends_on for endtoend to depend on zap.

Most of the time it will be fine but occasionally the race condition might mean the tests fail just because zap isn't up yet